### PR TITLE
Add ut for ShadowExtractor in shadow project

### PR DESCRIPTION
**Pull request by zzs52**
Fixes Add ut for ShadowExtractor in shadow project #12498.

Changes proposed in this pull request:
-
-
-

### CODE REVIEW

File: shardingsphere-features/shardingsphere-shadow/shardingsphere-shadow-core/src/test/java/org/apache/shardingsphere/shadow/route/engine/util/ShadowExtractorTest.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.junit.Assert.assertThat;
+
+public final class ShadowExtractorTest {
+
+    @Test
+    public void extractValuesIsSimpleExpressionSegment() {
+        LiteralExpressionSegment LiteralExpressionSegment = new LiteralExpressionSegment(1, 2, "");
+        List<Object> list = new LinkedList<>();
```
Comment (taojintianxia) [inline shardingsphere-features/shardingsphere-shadow/shardingsphere-shadow-core/src/test/java/org/apache/shardingsphere/shadow/route/engine/util/ShadowExtractorTest.java:36]: seems`list` is not a good variable name.

Comment (zzs52) [reply shardingsphere-features/shardingsphere-shadow/shardingsphere-shadow-core/src/test/java/org/apache/shardingsphere/shadow/route/engine/util/ShadowExtractorTest.java]: > seems`list` is not a good variable name.

Ok,I got it.

File: shardingsphere-features/shardingsphere-shadow/shardingsphere-shadow-core/src/test/java/org/apache/shardingsphere/shadow/route/engine/util/ShadowExtractorTest.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+import org.junit.Test;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.junit.Assert.assertThat;
+
+public final class ShadowExtractorTest {
+
+    @Test
+    public void extractValuesIsSimpleExpressionSegment() {
+        LiteralExpressionSegment LiteralExpressionSegment = new LiteralExpressionSegment(1, 2, "");
```
[the remark this answers is not in the document]
Comment (soulasuna) [reply shardingsphere-features/shardingsphere-shadow/shardingsphere-shadow-core/src/test/java/org/apache/shardingsphere/shadow/route/engine/util/ShadowExtractorTest.java:35]: Test case of `SimpleExpressionSegment`, must use `SimpleExpressionSegment` interface to test, the implementation class can be `LiteralExpressionSegment`，expected value should not be empty  Example: 

SimpleExpressionSegment simpleExpressionSegment = new LiteralExpressionSegment(1, 2, "expected");

File: shardingsphere-features/shardingsphere-shadow/shardingsphere-shadow-core/src/test/java/org/apache/shardingsphere/shadow/route/engine/util/ShadowExtractorTest.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+        List<Object> parametersList = new LinkedList<>();
+        Collection<Comparable<?>> result = new LinkedList<>();
+        result.add((Comparable<?>) LiteralExpressionSegment.getLiterals());
+        assertThat(ShadowExtractor.extractValues(LiteralExpressionSegment, parametersList), is(Optional.of(result)));
```
Comment (soulasuna) [inline shardingsphere-features/shardingsphere-shadow/shardingsphere-shadow-core/src/test/java/org/apache/shardingsphere/shadow/route/engine/util/ShadowExtractorTest.java:39]: AssertThat, please use the exact value，example:

```java
assertTrue(ShadowExtractor.extractValues(simpleExpressionSegment, parametersList).isPresent());
assertThat(ShadowExtractor.extractValues(simpleExpressionSegment, parametersList).get().iterator().next(), is("expected"));
```

**Review by soulasuna — COMMENTED** [review]
Thank you for your contribution.
Please fix them.

File: shardingsphere-features/shardingsphere-shadow/shardingsphere-shadow-core/src/test/java/org/apache/shardingsphere/shadow/route/engine/util/ShadowExtractorTest.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (zzs52) [reply shardingsphere-features/shardingsphere-shadow/shardingsphere-shadow-core/src/test/java/org/apache/shardingsphere/shadow/route/engine/util/ShadowExtractorTest.java:35]: > Test case of `SimpleExpressionSegment`, must use `SimpleExpressionSegment` interface to test, the implementation class can be `LiteralExpressionSegment`，expected value should not be empty Example:
> 
> ```java
> SimpleExpressionSegment simpleExpressionSegment = new LiteralExpressionSegment(1, 2, "expected");
> ```

Ok,thank you very much.

File: shardingsphere-features/shardingsphere-shadow/shardingsphere-shadow-core/src/test/java/org/apache/shardingsphere/shadow/route/engine/util/ShadowExtractorTest.java (hunk 3 above)
Comment (zzs52) [reply shardingsphere-features/shardingsphere-shadow/shardingsphere-shadow-core/src/test/java/org/apache/shardingsphere/shadow/route/engine/util/ShadowExtractorTest.java:39]: Ok,thank you very much.

**Comment by zzs52** [comment]
There are still two places that fail the CI test, and I don’t know how to solve it.Don't know if it will affect.

File: shardingsphere-features/shardingsphere-shadow/shardingsphere-shadow-core/src/test/java/org/apache/shardingsphere/shadow/route/engine/util/ShadowExtractorTest.java [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+import org.junit.Test;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.assertTrue;
+
+public final class ShadowExtractorTest {
+
+    @Test
+    public void extractValuesIsSimpleExpressionSegment() {
```
Comment (soulasuna) [inline shardingsphere-features/shardingsphere-shadow/shardingsphere-shadow-core/src/test/java/org/apache/shardingsphere/shadow/route/engine/util/ShadowExtractorTest.java:34]: The test method should start with `assert`.
Please change the method name to `assertExtractValuesSimpleExpressionSegment`.

**Review by soulasuna — COMMENTED** [review]
Please fix it.

File: shardingsphere-features/shardingsphere-shadow/shardingsphere-shadow-core/src/test/java/org/apache/shardingsphere/shadow/route/engine/util/ShadowExtractorTest.java (hunk 4 above)
Comment (zzs52) [reply shardingsphere-features/shardingsphere-shadow/shardingsphere-shadow-core/src/test/java/org/apache/shardingsphere/shadow/route/engine/util/ShadowExtractorTest.java:34]: Ok,thanks for reminding.

Comment (zzs52) [reply shardingsphere-features/shardingsphere-shadow/shardingsphere-shadow-core/src/test/java/org/apache/shardingsphere/shadow/route/engine/util/ShadowExtractorTest.java:34]: > The test method should start with `assert`. Please change the method name to `assertExtractValuesSimpleExpressionSegment`.

Ok,thanks for reminding.

**Review by soulasuna — APPROVED** [review]

**Review by menghaoranss — APPROVED** [review]

**Pull request merged**